hngprojects/hng_boilerplate_nestjs
# feat: update comment

**What does the PR do?**
This PR introduces new API endpoints to update comments. It includes server-side logic for handling comment update, along with appropriate validation and error handling.

How should this be manually tested?

To Update a Comment:

Send a PATCH request to /api/v1/comments/update/:comment_id with a valid UpdateCommentDto payload and a valid comment_id in the request.
Example:

`PATCH /api/v1/comments/update/:comment_id`

{
  "content": "This is the update to the original comment"
}

Successful Response

On a successful comment update, the API should return a 200 SUCCESS status code.

```json
{
      "status": "status_code",
      "message": "Comment updated successfully",
      "data": {
            "id": "84488317-234b-41b5-a2d1-07d47a50be8e",
            "created_at": "2024-08-08T06:27:40.309Z",
            "updated_at": "2024-08-08T06:59:59.030Z",
            "comment": "I just update this comment now",
            "model_id": "1234",
            "model_type": "post"
            }
      "commentedBy" : "John Smith"
}
```

Error Handling:

When the current user is not authourized:

```json
{
          "status": "error",
          "message": "User not authorized",
          "status_code": "401",
 }
```


When the current user is not found:

```json
{
          "status": "error",
          "message": "User nof found",
          "status_code": "404",
 }
```

Error During coment update

If the comment creation is not successful:

```json
{
  "status": "Error",
  "message": "Error creating comment",
  "status_code": 500
}
```

 **Checklist of what has been done**
- [X ] Define API endpoint to update comment.
- [X ]  Validate and handle authentication token.
- [ X]  Implement logic to update a comment.
- [ X]  Handle success and error responses.
- [ X]  Test Successful response

**API Test Screenshot**
![Screenshot from 2024-08-08 07-59-37](https://github.com/user-attachments/assets/41665ec6-034f-484d-8558-d794986509e6)

** Jest Test Screenshot **
![Screenshot from 2024-08-07 15-46-24](https://github.com/user-attachments/assets/19fbf8e7-bc34-474c-b6c1-2623553258d3)

Referenced Linear Ticket: [Ticket-Link](https://linear.app/team-bingo/issue/BAC2-134/endpoint-to-edit-comment-or-212)
Referenced Issue: [Link to Issue](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/212)
